### PR TITLE
Add Docsearch with credentials

### LINKED ISSUE #15: Add Search to docs
### Description
Add a search bar for internal docs search.

### Additional information
https://docusaurus.io/docs/search

### PR DESCRIPTION
**Pull request by TalhaMaliktz**
### What was the problem?

This PR resolves #15

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -77,6 +77,35 @@ const config = {
...
+        contextualSearch: true,
+
+        // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
+        externalUrlRegex: 'external\\.com|domain\\.com',
+
+        // Optional: Replace parts of the item URLs from Algolia. Useful when using the same search index for multiple deployments using a different baseUrl. You can use regexp or string in the `from` param. For example: localhost:3000 vs myCompany.com/docs
+        replaceSearchResultPathname: {
+          from: '/docs/', // or as RegExp: /\/docs\//
+          to: '/',
+        },
```
Comment (Tschakki) [inline docusaurus.config.js]: Do we really need this?

Comment (TalhaMaliktz) [reply docusaurus.config.js]: I think so! Copied it from here: https://docusaurus.io/docs/search#connecting-algolia

Comment (Tschakki) [reply docusaurus.config.js]: Yeah, but isn't is stated in the comments, that these are optional configuration, for the case if there are multiple deployments?

Comment (TalhaMaliktz) [reply docusaurus.config.js]: I have removed all the optional properties and have only added the properties mentioned in the official email received from Algolia. Please check now.

**Review by Tschakki — APPROVED** [review]
LGTM 🚀

**Pull request merged**